### ISSUE TITLE
vscode extension doesn't show any diagnostics

### ISSUE THREAD

**Issue by DetachHead** [gauge-sh/tach#424]
the vscode extension does not show any diagnostics, even though the CLI does

# project files
```toml
# ./tach.toml

exclude = [
  ".*__pycache__",
  ".*egg-info",
  "docs",
  "tests",
]
source_roots = [
  ".",
]

[[modules]]
path = "sandpit.foo"
depends_on = []
[[modules]]
path = "sandpit.bar"
depends_on = []
```

```jsonc
// ./.vscode/settings.json
{
    "tach.importStrategy": "fromEnvironment"
}
```
```py
# ./sandpit/foo.py

from sandpit.bar import a
```
```py
# ./sandpit/bar.py

a = 1
```
# cli output
```
> tach check
❌ sandpit\foo.py[L1]: Cannot import 'sandpit.bar.a'. Module 'sandpit.foo' cannot depend on 'sandpit.bar'. 
```
# vscode output
## problems tab
![image](https://github.com/user-attachments/assets/f4c35c69-61ac-457c-8f2c-d05ad31b43b8)

## "Tach" log from the output tab
```
2024-11-19 10:17:59.432 [info] [Trace - 10:17:59 AM] Sending notification 'textDocument/didChange'.
2024-11-19 10:17:59.686 [info] [Trace - 10:17:59 AM] Sending notification 'textDocument/didChange'.
2024-11-19 10:17:59.784 [info] [Trace - 10:17:59 AM] Sending notification 'textDocument/didSave'.
2024-11-19 10:17:59.790 [info] [Trace - 10:17:59 AM] Received notification 'window/logMessage'.
2024-11-19 10:17:59.790 [info] CWD Linter (no rpc/doc): c:\Users\user\project
2024-11-19 10:18:02.915 [info] [Trace - 10:18:02 AM] Received notification 'window/logMessage'.
2024-11-19 10:18:02.915 [info] file:///c%3A/Users/user/project/sandpit/foo.py :
<tach.extension.CheckDiagnostics object at 0x000001B91A6CC2D0>
2024-11-19 10:18:02.915 [info] [Trace - 10:18:02 AM] Received notification 'textDocument/publishDiagnostics'.
```
# environment info
tach 0.14.3
python 3.13
windows 11

**Comment by caelean** [comment]
![image](https://github.com/user-attachments/assets/1240e123-1e35-46d6-ab22-6d047f5589b8)

This case is working for me, although I'm not on a windows machine. I'll try and get on one and re-test.

One follow up question @DetachHead - is `tach.toml` in the root of your VSCode's workspace? I have an outstanding TODO to allow configuration of the location of `tach.toml` as right now there's an assumption in place related to the workspace configuration.

CC @KotlinIsland @scott-au, if you're also experiencing this, curious as to your OS / workspace path vs. tach.toml path.

**Comment by DetachHead** [comment]
> One follow up question @DetachHead - is `tach.toml` in the root of your VSCode's workspace?

yeah it is

**Comment by emdoyle** [comment]
@DetachHead we just shipped a new version of the VSCode extension (0.18.0) which incorporates significant changes:

https://github.com/gauge-sh/tach/pull/474
https://github.com/gauge-sh/tach/pull/484
https://github.com/gauge-sh/tach-vscode/pull/15

Could you try upgrading to the latest version of the extension and seeing if diagnostics are still missing? The latest VSCode extension requires the latest Tach (>=0.18.0), but the bundled version should also now be working if you don't want to upgrade Tach in your dev environment.

**Comment by DetachHead** [comment]
updated both the version in my dev environment and the extension and the diagnostic still don't seem to be appearing unfortunately

**Comment by emdoyle** [comment]
> updated both the version in my dev environment and the extension and the diagnostic still don't seem to be appearing unfortunately

Ok sorry about the back and forth, I'm going to see if I can replicate in a Windows VM - have a guess about this being some kind of path comparison issue but would want to validate a fix before shipping another release. Will let you know when there is a fix to try!

**Comment by DetachHead** [comment]
no worries, i know it's a pain to deal with issues like this

**Comment by emdoyle** [comment]
@DetachHead There is a fix for this in `0.19.1` which was just released, please let me know if it works for you! I repro'd the issue on a Windows VM and this change resolves the issue there.

A new version of the extension with Tach 0.19.1 bundled will be released shortly.

**Comment by DetachHead** [comment]
works now, thanks!
